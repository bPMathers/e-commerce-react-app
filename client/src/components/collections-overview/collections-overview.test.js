import React from 'react';
import { shallow } from 'enzyme'

import { CollectionsOverview } from './collections-overview.component';

describe('CollectionsOverview component', () => {
  let wrapper, mockProps;

  beforeEach(() => {
    mockProps = {
      collections: [{
        id: 1,
        title: 'hats',
        items: []
      },
      {
        id: 2,
        title: 'shoes',
        items: []
      },
      ]
    }
    wrapper = shallow(<CollectionsOverview {...mockProps} />);
  });

  it('should render CollectionsOverview component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the same number of nested CollectionsPreview as there are passed as props', () => {
    expect(wrapper.find('CollectionsOverviewContainer').children().length).toEqual(mockProps.collections.length);
  })
});
