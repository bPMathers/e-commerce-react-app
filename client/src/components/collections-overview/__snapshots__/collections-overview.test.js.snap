// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionsOverview component should render CollectionsOverview component correctly 1`] = `
<CollectionsOverviewContainer>
  <withRouter(CollectionPreview)
    items={Array []}
    key="1"
    title="hats"
  />
  <withRouter(CollectionPreview)
    items={Array []}
    key="2"
    title="shoes"
  />
</CollectionsOverviewContainer>
`;
